### PR TITLE
Fix for module installation with db prefixes

### PR DESCRIPTION
**Pull request by bartoszkubicki**
Fixing process + removal the only attachement using product form + url-type attachment download

### CODE REVIEW

**Comment by bartoszkubicki** [comment]
Solves #5

File: Controller/Download/Attachment.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-declare(strict_types = 1);
+declare(strict_types=1);
```
Comment (krzysztofkuzniar) [inline Controller/Download/Attachment.php]: Sprawdź wszystkie pliki *.php i popraw wszędzie `declare` tak jak tutaj

File: Controller/Download/Attachment.php [hunk 2]
```diff
@@ -63,24 +62,23 @@ public function __construct(
         $this->fileHelper = $fileHelper;
     }
 
-
     /**
-     * @return \Magento\Framework\Controller\ResultInterface
+     * @return ResultInterface|ResponseInterface
      */
-    public function execute() : ResultInterface
+    public function execute()
```
Comment (krzysztofkuzniar) [inline Controller/Download/Attachment.php]: czy tutaj zasadne/potrzebne jest usunięcie typu zwracanego?

File: Plugin/InitForSave.php [hunk 3]
```diff
@@ -56,41 +54,47 @@ public function __construct(
...
-     *
-     * @throws \Magento\Framework\Exception\LocalizedException
-     *
-     * @return \Magento\Catalog\Model\Product
+     * @param Helper $subject
+     * @param Product $product
+     * @return Product
+     * @throws LocalizedException
      */
     public function afterInitialize(Helper $subject, Product $product)
```
Comment (krzysztofkuzniar) [inline Plugin/InitForSave.php]: return type

File: Plugin/InitForSave.php [hunk 4]
```diff
@@ -56,41 +54,47 @@ public function __construct(
...
     {
-        if ($downloadable = $this->request->getPost('downloadable')) {
-            if (isset($downloadable['attachment']) && is_array($downloadable['attachment'])) {
-                $product->setDownloadableData($downloadable);
-                $extension = $product->getExtensionAttributes();
+        $downloadable = $this->request->getPost('downloadable');
+
+        if (!empty($downloadable) && (isset($downloadable['attachment']) && is_array($downloadable['attachment']))) {
```
Comment (krzysztofkuzniar) [inline Plugin/InitForSave.php]: te sprawdzenia, porównania dałbym do osobnej metody

File: Plugin/InitForSave.php [hunk 5]
```diff
@@ -56,41 +54,47 @@ public function __construct(
...
-                        $attachments[] = $this->attachmentBuilder->setData(
-                            $attachmentData
-                        )->build(
-                            $this->attachmentFactory->create()
-                        );
-                    }
+            $attachments = [];
+            foreach ($downloadable['attachment'] as $attachmentData) {
+                if (!$attachmentData ||
+                    (isset($attachmentData['is_delete']) && (bool) $attachmentData['is_delete'])) {
```
Comment (krzysztofkuzniar) [inline Plugin/InitForSave.php]: tutaj podobnie, dałbym to w osobną metodę

File: Controller/Download/Attachment.php (hunk 1 above)
Comment (bartoszkubicki) [reply Controller/Download/Attachment.php]: All refactor-like notes will be corrected in separate PR, as whole module needs to be refactored (it is quite old). I will skip it in scope of this PR.

File: Plugin/InitForSave.php (hunk 4 above)
Comment (bartoszkubicki) [reply Plugin/InitForSave.php]: All refactor-like notes will be corrected in separate PR, as whole module needs to be refactored (it is quite old). I will skip it in scope of this PR.

File: Plugin/InitForSave.php (hunk 5 above)
Comment (bartoszkubicki) [reply Plugin/InitForSave.php]: All refactor-like notes will be corrected in separate PR, as whole module needs to be refactored (it is quite old). I will skip it in scope of this PR.

File: Controller/Download/Attachment.php (hunk 2 above)
Comment (bartoszkubicki) [reply Controller/Download/Attachment.php]: This deletion solves issue, when attachement is of type url. For now more complex refactor of this method, allowing usage of return type is not worth, as general refactor of module is neeeded.

File: Plugin/InitForSave.php (hunk 3 above)
Comment (bartoszkubicki) [reply Plugin/InitForSave.php]: All refactor-like notes will be corrected in separate PR, as whole module needs to be refactored (it is quite old). I will skip it in scope of this PR.

**Pull request merged**